### DISCUSSION TITLE
Some System properties seem to be ignored in final build prism.allowhidpi

### DISCUSSION THREAD

**Discussion by sproket** [fvarrui/JavaPackager#442] [Q&A]
Hi,

For my Windows build I discovered an issue if the user sets scaling to 125% - I don't want my game to scale.  Normally I just set the system property in my Launcher class which works fine in the IDE but once I make a build this seems to be ignored somehow,

Any ideas?

`public class Launcher {

    public static void main(String[] args) {

        System.setProperty("prism.allowhidpi", "false");
        GoldchestApp.main(args);
    }
}
`

`<plugin>
                <groupId>io.github.fvarrui</groupId>
                <artifactId>javapackager</artifactId>

                    <version>1.7.6</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>package</goal>
                        </goals>
                        <configuration>

                            
                            <mainClass>game.ui.Launcher</mainClass>
                            
                            <bundleJre>true</bundleJre>
                            <customizedJre>true</customizedJre>
                            <generateInstaller>true</generateInstaller>
                            <administratorRequired>false</administratorRequired>
                            <platform>auto</platform>
                            <displayName>${project.name}</displayName>
                            <vmArgs>
                                <vmArg>-ea</vmArg>
                                <vmArg>-XX:+UseZGC</vmArg>
                                <vmArg>-Dprism.allowhidpi=false</vmArg>
                            </vmArgs>

                            <winConfig>
                                
                                <headerType>console</headerType>
                                <icoFile>${basedir}/src/main/resources/rpgicon.ico</icoFile>
                                <exeCreationTool>winrun4j</exeCreationTool>
                                <wrapJar>false</wrapJar>
                                <generateSetup>true</generateSetup>
                                <generateMsi>false</generateMsi>
                                <generateMsm>false</generateMsm>

                                
                                <companyName>none so far....</companyName>
                                <fileVersion>${project.version}</fileVersion>
                                <txtFileVersion>${project.version}</txtFileVersion>
                                <productVersion>${project.version}</productVersion>
                                <txtProductVersion>${project.version}</txtProductVersion>
                                <fileDescription>${description}</fileDescription>
                                <copyright>Me 2023</copyright>
                                <productName>${project.name}</productName>
                                <internalName>${project.name}</internalName>
                                <originalFilename>${project.name}.exe</originalFilename>

                                <setupMode>installForCurrentUser</setupMode>
                                <disableDirPage>true</disableDirPage>
                                <disableProgramGroupPage>true</disableProgramGroupPage>
                                <disableFinishedPage>false</disableFinishedPage>
                                <disableRunAfterInstall>false</disableRunAfterInstall>
                                <disableWelcomePage>true</disableWelcomePage>
                                <createDesktopIconTask>true</createDesktopIconTask>
                                <removeOldLibs>true</removeOldLibs>
                            </winConfig>

                            <macConfig>
                                <generateDmg>true</generateDmg>
                                <generatePkg>false</generatePkg>

                                
                                <backgroundImage>${basedir}/src/main/resources/portraits/splash.png</backgroundImage>

                                <iconX>20</iconX>
                                <iconY>20</iconY>
                                <appsLinkIconX>100</appsLinkIconX>
                                <appsLinkIconY>20</appsLinkIconY>

                                <volumeName>${project.name}-Game</volumeName>
                            </macConfig>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
`

**Comment by fvarrui** [comment]
> ```java
> public static void main(String[] args) {
> 
>     System.setProperty("prism.allowhidpi", "false");
>     GoldchestApp.main(args);
> }
> ```

This doesn't work either? 🙄

**Reply by sproket** [reply]
No it doesn't. I have that in my main Launcher class.

It's easy to test if you have Windows. Just make a simple JavaFX app and try it with Scaling at 125%.

I also tried adding it to vmargs. No luck.  Might be an issue with winrun4j not sure yet.

**Reply by fvarrui** [reply]
> I also tried adding it to vmargs. No luck. Might be an issue with winrun4j not sure yet.

Ok, as you mention in #443, it seems to be an issue with WinRun4J.
